src/controllerSynonyms.ts

```diff
@@ -0,0 +1,226 @@
+ }: {
+    _postgres: Pool,
+    _selectAllSynonymsFromDb?: typeof selectAllSynonymsFromDb
+  }): Promise<any[]> {
+
+  const result =  await _selectAllSynonymsFromDb({ _postgres });
+  const reduceToListSynonyms = R.pick(['list_synonyms']);
+
+  return R.pipe(R.map(reduceToListSynonyms), R.map(R.values))(result);
+}
+
+// --------------------------
+
+function selectSynonymsFromDb({
+  _postgres,
+  _symbol
+}: {
+    _postgres: Pool,
+    _symbol: string
+  }) {
+  const query = {
+    text:
+    `SELECT list_synonyms, last_update
+    FROM synonyms
+    INNER JOIN symbols
+    ON synonyms.id = symbols.synonyms_uid
+    WHERE symbols.symbol = $1`,
+    values: [_symbol]
+  };
+
+  const result = _postgres.query(query)
+    .then(data => data.rows);
+
+  return result;
+}
+
+// --------------------------
+
+function selectAllSynonymsFromDb({
+  _postgres
+}: {
+    _postgres: Pool
+  }) {
+  const query = {
+    text:
+    `SELECT list_synonyms FROM synonyms`
+  };
+
+  const result = _postgres.query(query)
+  .then(data => data.rows);
+
+  return result;
+}
+
+// ------------------------
+
+export async function updateSynonyms({
+  payload,
+  _postgres
+  }: {
+    payload: { symbol: string },
+    _postgres: Pool
+  }) {
+
+  // get synonyms from HUGO
+  const listSynonyms = await synonyms({ payload });
+  storeSynonyms({ listSynonyms, _postgres });
+}
+
+// ------------------------
+
+export async function populateSynonyms({
+  _postgres,
+  _deleteAllSynonymsFromDb = deleteAllSynonymsFromDb,
+  _storeSynonyms = storeSynonyms
+  }: {
+    _postgres: Pool,
+    _deleteAllSynonymsFromDb?: typeof deleteAllSynonymsFromDb,
+    _storeSynonyms?: typeof storeSynonyms
+  }) {
+
+  // get synonyms from HUGO
+  _deleteAllSynonymsFromDb({ _postgres });
+  const listlistSynonyms = await allSynonyms();
+  listlistSynonyms.forEach((listSynonyms: any) => {
+    _storeSynonyms({ listSynonyms, _postgres });
+  });
+  console.log('Database populated!');
+}
+
+// --------------------------
+async function storeSynonyms({
+  listSynonyms,
+  _postgres
+  }: {
+    listSynonyms: any,
+    _postgres: Pool
+  }) {
+
+  if (listSynonyms && listSynonyms.length >= 1) {
+    const convertedListSynonyms = JSON.stringify(listSynonyms).replace('[','\[').replace(']','\]').replace('\'','');
+    const insertSynonyms= {
+      text: 'INSERT INTO synonyms(list_synonyms, last_update) VALUES($1, CURRENT_TIMESTAMP) RETURNING * ',
+      values: [convertedListSynonyms]
+    };
+
+    // insert standardised terms into table, return the row id to link to original term
+    const synonyms_uid = await _postgres.query(insertSynonyms)
+      .then((data: any) => data.rows[0].id)
+      .catch(function(e) {
+        return -1;
+      });
+
+    for ( const synonymToken of listSynonyms ) {
+      const createQuery = {
+        text: 'INSERT INTO symbols(synonyms_uid, symbol) VALUES($1, $2)',
+        values: [synonyms_uid, synonymToken]
+      };
+
+      _postgres.query(createQuery);
```

Comment:
If you do not handle the promise rejection it will blow up in production.

Reply:
Done.